addons/stock/models/product.py

```diff
@@ -886,6 +886,23 @@ def _onchange_type(self):
             }
         return res
 
+    @api.onchange('uom_id')
+    def _onchange_uom_id(self):
+        done_moves = self.env['stock.move'].sudo().search_count(
```

Comment:
Nitpick, but I'm not sure if we're looking after `done` moves. If we do, then you need to add it to the domain. If not, then just call it `moves`, as I fear it may raise confusion in the future.